app/api/getProducts/route.ts

```diff
@@ -0,0 +1,49 @@
+// app/api/post/route.ts
+import postgres from '@/app/utils/postgres';
+import { NextRequest, NextResponse } from 'next/server';
+
+export async function POST(request: NextRequest) {
+    try {
+        const client = await postgres.connect();
+
+        const { productName, itemCode, quantity: rawQuantity } = await request.json();
+        const quantity = rawQuantity ? parseInt(rawQuantity) : null;
+
+        if (!productName || !itemCode || !quantity || quantity < 0) {
```

Comment:
Is this a validation check for required fields? I think there is a required attribute in the input tag. Perhaps you can use that.